From 09432c7aa56157f626b1a1d44c3f5fa33d843d36 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:53 -0400
Subject: [ethernet] i40e: enable descriptor prefetch for VFs

Message-id: <1409658042-22689-113-git-send-email-sassmann@redhat.com>
Patchwork-id: 90308
O-Subject: [RHEL7.1 PATCH 112/281] i40e: enable descriptor prefetch for VFs
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

As recommended by the hardware guys, enable descriptor prefetch for
rings belonging to VFs. This matches a change already made for ring
belonging to the PF.

Change-ID: Idcc4dc7064bc9144ea81f5701ad07b7ecd11ba49
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 50d41659fac469fc10eaad594baa8f1d8b09c5a1)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
index 4d21956..6916ac0 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_virtchnl_pf.c
@@ -354,6 +354,7 @@ static int i40e_config_vsi_rx_queue(struct i40e_vf *vf, u16 vsi_idx,
  rx_ctx.tphhead_ena = 1;
  rx_ctx.lrxqthresh = 2;
  rx_ctx.crcstrip = 1;
+ rx_ctx.prefena = 1;
 
  /* clear the context in the HMC */
  ret = i40e_clear_lan_rx_queue_context(hw, pf_queue_id);
-- 
1.7.1